internal/provider/azure/config.go

```diff
@@ -82,6 +82,14 @@ func (p azureEnvironProvider) ConfigDefaults() schema.Defaults {
 	return configDefaults
 }
 
+// ModelConfigDefaults provides a set of default model config attributes that
+// should be set on a models config if they have not been specified by the user.
+func (prov *azureEnvironProvider) ModelConfigDefaults(_ context.Context) (map[string]any, error) {
```

Comment:
With this here, is there a reason why the corresponding code in `PrepareConfig` isn't removed?
There's probably an argument that PrepareConfig could be designed out now. It existed primarily to add in provider specific config to the model, which is essentially what this PR does. It also does validation of the cloud spec but I would argue that's a conflation of responsibilities.

```
	// PrepareConfig prepares the configuration for a new model, based on
	// the provided arguments. PrepareConfig is expected to produce a
	// deterministic output. Any unique values should be based on the
	// "uuid" attribute of the base configuration. This is called for the
	// controller model during bootstrap, and also for new hosted models.
	PrepareConfig(context.Context, PrepareConfigParams) (*config.Config, error)
```

Reply:
You are correct in every respect above. I want to remove it but I am trying to avoid busy work as this change is a little hard to pull out because of the client.

Reply:
Can we add a TODO on the method declaration in the interface. At the least, we should follow up immediately with a PR to remove the unnecessary config manipulation, even if the cloud spec validation remains for now.